sdk/samples/iot/pnp/pnp_protocol.c

```diff
@@ -351,18 +364,20 @@ az_result pnp_process_device_twin_message(
 
       property_name = jr.token;
 
-      if (az_result_failed(az_json_reader_next_token(&jr)))
+      rc = az_json_reader_next_token(&jr);
+      if (az_result_failed(rc))
       {
         IOT_SAMPLE_LOG_ERROR("Failed to get next token.");
         return AZ_ERROR_UNEXPECTED_CHAR;
       }
 
-      if (jr.token.kind == AZ_JSON_TOKEN_BEGIN_OBJECT && components_ptr != NULL
-          && (az_result_succeeded(
-              is_component_in_model(property_name.slice, components_ptr, components_num, &index))))
+      rc = is_component_in_model(property_name.slice, components_ptr, components_num, &index);
+      if ((jr.token.kind == AZ_JSON_TOKEN_BEGIN_OBJECT) && (components_ptr != NULL)
```

Comment:
Why is the `(jr.token.kind == AZ_JSON_TOKEN_BEGIN_OBJECT)` happening **after** `is_component_in_model`. That seems unrelated to JSON. Should it be before?

Reply:
@danewalton thoughts?  I will take a look at as well.

Reply:
No so the marker of a component is if the property name is equal to a registered component and if its value is a JSON object (`AZ_JSON_TOKEN_BEGIN_OBJECT`). So it first checks to see if the prop name is a component and if it is, it makes sure the next token is a `AZ_JSON_TOKEN_BEGIN_OBJECT`.